### PR TITLE
[TST] Refactor tests

### LINKED ISSUE #126: Write a test for behaviour of `RETURN_AGG` option
Currently, several of the tests in test_query.py use an aggregate query results object (`test_data()`) for testing.

We don't currently have a test that checks the structure of query results returned from the API depending on the set value of `RETURN_AGG`.

### Steps to implement
We could do one or both of
- ~testing the `create_query` function to check that the SPARQL query is wrapped with an aggregator statement when `RETURN_AGG=true`~
- mocking a small response from the graph (monkeypatching the `httpx` post function) to see if the graph results are parsed appropriately on the API side depending on `RETURN_AGG`

### LINKED ISSUE #266: Refactor mock get method for testing to accept return object
We keep using a `mocked_get()` function pretty much everywhere. A variant of this has already been refactored into the `conftest.py` file, so it probably makes sense to just have a single mock method but have it accept the return type as an argument. Something like so:

```python
def my_mock_get(expectedReturn):
  async def inner_mock(some, other, args):
    return expectedReturn
  return inner_mock
```

It seems that you can do something like this with fixtures in pytest using indirect parameterizations: https://docs.pytest.org/en/latest/example/parametrize.html#apply-indirect-on-particular-arguments

_Originally posted by @surchs in https://github.com/neurobagel/api/pull/264#discussion_r1462460857_

### PR DESCRIPTION
**Pull request by alyssadai**
- Closes #126
- Closes #266 

Changes proposed in this pull request:

- Basic test added for structure of cohort query response when `RETURN_AGG=true`
- Turned mocked get function used across query tests into a few different fixtures, with variations based on the type of value/object to be returned by the mocked function
  - Specific return objects for the mocked function specified in test definitions via [indirect parametrization](https://docs.pytest.org/en/latest/example/parametrize.html#indirect-parametrization)
  - `mock_successful_get` (which depends on another fixture for the return object) kept a separate function to avoid overly complicated parametrization when writing tests (i.e., you cannot easily pass one fixture as a parameter to another fixture)
 

## Checklist

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see https://neurobagel.org/contributing/pull_requests for more info)_
- [x] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -138,7 +152,27 @@ async def mockreturn(
         assessment,
         image_modal,
     ):
-        return None
+        return request.param
+
+    return mockreturn
```
Comment (surchs) [inline tests/conftest.py:157]: neat :tada:

File: tests/test_query.py [hunk 2]
```diff
@@ -114,6 +113,7 @@ def test_get_valid_age_single_bound(
     assert response.json() != []
 
 
+@pytest.mark.parametrize("mock_get", [None], indirect=True)
```
Comment (surchs) [inline tests/test_query.py:116]: phew. That's cool, but I see what you mean, bit tricky to "just read". I guess what this tells me is that you tell `mock_get` to have a `None` return here?

Comment (surchs) [reply tests/test_query.py:116]: I wonder if we haven't gone a little over-complex here. Pytest seems to have much simpler ways to pass data to fixtures from tests: https://docs.pytest.org/en/6.2.x/fixture.html#using-markers-to-pass-data-to-fixtures . I believe that would be a lot more readable. Wdyt?

**Review by surchs — APPROVED** [review]
Very cool PR @alyssadai, thanks for figuring all of this out.

Looking over the double parameterization here is impressive, but also a bit confusing:
1. when I read the parameterization I wonder: what is parameterized? There's only one parameter
2. when I read the test itself where I am using the mock I wonder: what does this mock actually do? 

I think we could probably live with that, but my sense is that pytest offers a simpler way to achieve the same outcome: https://docs.pytest.org/en/6.2.x/fixture.html#using-markers-to-pass-data-to-fixtures 

Maybe you could take a look at that section (and the "Factories as fixtures" just after it) and see if you agree. 

Otherwise: :cook:

File: tests/test_query.py (hunk 2 above)
Comment (alyssadai) [reply tests/test_query.py:116]: After looking into this, sticking with the original indirect parametrization approach seems to be more intuitive in terms of syntax and from a test-writing perspective, while still helping us avoid re-mocking get functions inside individual tests. So proceeding with the original changes!

**Comment by surchs** [comment]
:rocket: PR was released in [`v0.2.0`](https://github.com/neurobagel/api/releases/tag/v0.2.0) :rocket:

**Pull request merged**